Automattic/jetpack
# Site Logo: update for compatibility with WP 5.8 site_logo option

#### Changes proposed in this Pull Request:

Updates the legacy site-logo feature to use an attachment ID for the `site_logo` option, rather than a settings array. This makes the feature compatible with the Site Logo block, which is being included in the WP 5.8 release schedule for July 20.

Note that there are wpcom themes that access the `site_logo` option directly, so these will need updates for the new format.

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

n/a

#### Testing instructions:

_I recommend testing with current WP release (5.7.2) and with Gutenberg plugin deactivated, initially, until a [current bug with the Site Logo block](https://github.com/Automattic/wp-calypso/issues/54055#issuecomment-869953902) in Gutenberg is fixed._

* Before switching to this branch, activate a theme that uses the site-logo feature and save a logo in the customizer. For example: the WP.com versions of Soho, Libre, or Candela (you'll need to pull these from a wpcom sandbox, _not_ the dotorg themes repo).
* Notice that the `site_logo` option is saved as an array
* Checkout this branch of Jetpack and load the site... the `site_logo` option should be converted to an attachment ID and the logo should still display correctly.
* Refresh the Customizer, and see the new Core WP control now used for site logo, including a cropping step using the size recommend by the theme.
* Try removing the logo.
* Try saving a new logo in the Customizer. (you may not see the preview update, if there was no logo set before)
* Try switching to a theme that uses the `custom-logo` feature, like TwentyTwenty. The logo you set should be transfered to that theme (if it doesn't already have a logo set in the theme_mods).

**Testing wpcom Simple Sites**

Use D63465-code

- Sandbox both the site domain and public-api
- Use a site with a premium plan or business plan (non-Atomic)
- Follow the same testing instructions as above
